account_invoice_inter_company/tests/test_inter_company_invoice.py

```diff
@@ -2,37 +2,38 @@
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 
 from odoo import _
-from odoo.tests.common import TransactionCase
+from odoo.tests.common import SavepointCase
 from odoo.exceptions import ValidationError
+from odoo.modules.module import get_resource_path
+from odoo.tools import convert_file
 
 
-class TestAccountInvoiceInterCompany(TransactionCase):
-    def setUp(self):
-        super(TestAccountInvoiceInterCompany, self).setUp()
-        self.wizard_obj = self.env['wizard.multi.charts.accounts']
-        self.account_obj = self.env['account.account']
-        self.invoice_obj = self.env['account.invoice']
-        self.invoice_company_a = self.env.ref(
+class TestAccountInvoiceInterCompany(SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super(TestAccountInvoiceInterCompany, cls).setUpClass()
+        module = "account_invoice_inter_company"
+        convert_file(
+            cls.cr, module,
+            get_resource_path(module, "tests", "inter_company_invoice.xml"),
+            None, 'init', False, 'test', cls.registry._assertion_report,
+        )
+        cls.wizard_obj = cls.env['wizard.multi.charts.accounts']
+        cls.account_obj = cls.env['account.account']
+        cls.invoice_obj = cls.env['account.invoice']
+        cls.invoice_company_a = cls.env.ref(
             'account_invoice_inter_company.customer_invoice_company_a')
-        self.user_company_a = self.env.ref(
+        cls.user_company_a = cls.env.ref(
             'account_invoice_inter_company.user_company_a')
-        self.user_company_b = self.env.ref(
+        cls.user_company_b = cls.env.ref(
             'account_invoice_inter_company.user_company_b')
 
-        self.chart = self.env['account.chart.template'].search([], limit=1)
-        if not self.chart:
+        cls.chart = cls.env['account.chart.template'].search([], limit=1)
```

Comment:
This shouldn't be the method. You have a method `try_load_chart_template` or similar executed on company record for loading the proper chart template.

Reply:
AFAICS this line can be removed, as it's not used. I don't want to refactor everything unless strictly necessary, TBH.

Reply:
But you need to load the chart template any way. Previously they tricked the system adding as demo all the stuff that this method makes. You need also to look for demo XML-IDs used here in the test.

Reply:
Yes, I'm just loading the XML data in the test itself. [It seems a common pattern for testing](https://github.com/odoo/odoo/search?l=Python&q=def+_load) when you need to lazy-load some demo data, or you need to ensure its state. This way I don't need to rewrite the tests.

You can see [it works fine](https://travis-ci.org/OCA/multi-company/jobs/428578869#L1213).

Reply:
Well, it can cause side effects, as not all data is properly set (from the `account` module perspective, and this is an irregular way of having a chart of account. For example, field `chart_template_id` on company is not set - as there's no chart template -), so this can throw side effects, not right now, but on future changes or dependent modules.